test-case/multiple-pipeline.sh

```diff
@@ -8,22 +8,16 @@
 ## Description:
 ##    Run multiple pipelines in parallel
 ##    Rule:
-##      1. playback first mode: playback pipelines will be started first, if playback
-##         pipeline count is less than requested pipeline count defined by -c option,
-##         capture pipelines will be started to ensure requested number of pipelines are
-##         running in parallel.
-##      2. capture first mode: same with playback mode, but capture pipelines will be
-##         started first.
-##      3. all pipelines mode: run all pipelines in parallel by
-##          a. use option '-f a' OR
-##          b. specify a very big requested pipeline count to -c option
+##      1. Playback mode: playback pipelines ONLY are used, test pipeline count
+##         (OPT_VAL['c']) is respected, but maxed out with playback pipelines.
+##      2. Capture mode: same with playback mode, but capture pipelines ONLY will
+##         be used.
+##      3. All pipelines mode: Run all playback and capture pipelines in parallel
+##         by using the option '-f a' and set the number of pipelines with -c.
+##         If you need to use all pipelines, set a high pipeline count such as -c 20.
```

Comment:
what happens if the -c option is lower than the total of pipelines?

I also feel that this option could quickly run in trouble, with e.g. the nocodec topology we have multiple mixers and multiple copiers on capture, so we could end-up with 10+ streams in parallel. That's likely to fail in practice.

We may need to have two counts for capture and playback each, it's more pragmatic to me than trying to run 20 pipelines.

Reply:
If -c option has less than total pipeline. first fill with playback pipelines, rest of number will be filled with capture. so capture pipeline will be limited by the number.

For example, with -c 20, MTLP NOCODEC tplg has playback 4 pcm + capture 7 pcm = total 11 pcm. so mulitple-pipeline-all will start all 11 pcm at same time. If -c 8 is set, playback 4 pcm + capture 4 pcm will be used. If any pcm need to be skipped, there is a way to filter out by pcm name.

If we have two counts, there needs to be a little more condition checking between two counts. still trivial but I need more requirement for all(playback + capture) case.